test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/AssemblyPartTest.cs

```diff
@@ -22,6 +22,20 @@ public void AssemblyPart_Name_ReturnsAssemblyName()
         }
 
         [Fact]
+        public void AssemblyPart_Types_ReturnsDefinedTypes()
+        {
+            // Arrange
+            var assembly = typeof(AssemblyPartTest).GetTypeInfo().Assembly;
+            var part = new AssemblyPart(assembly);
+
+            // Act
+            var types = part.Types;
+
+            // Assert
+            Assert.Equal(assembly.DefinedTypes, types);
```

Comment:
Ah, yeah, not sure what xUnit does - you're probably right. But indeed perhaps comparing against a known good list of types is best here.